src/modules/auth/auth.service.ts

```diff
@@ -82,13 +89,77 @@ export class AuthService {
     const accessToken = await this.generateAuthToken(payload, 'access');
     const refreshToken = await this.generateAuthToken(payload, 'refresh');
 
+    const refreshTokenEntity = this.createRefreshTokenEntity(refreshToken);
+    await this.refreshTokenRepository.insert(refreshTokenEntity);
+
     return {
       message: 'User logged in successfully',
       accessToken,
       refreshToken,
     };
   }
 
+  async issueNewTokens(refreshToken: string) {
+    let decodedRefreshToken: AuthTokenPayload;
+
+    try {
+      decodedRefreshToken = await this.jwtService.verifyAsync<AuthTokenPayload>(
+        refreshToken,
+        { secret: this.configService.get<string>('JWT_REFRESH_TOKEN_SECRET') },
+      );
+    } catch (error: unknown) {
+      if (error instanceof TokenExpiredError) {
+        decodedRefreshToken =
```

Comment:
Don't understand what's happening here. Why we are still doing something for expired token?

Reply:
I was updating its deleted status, but will now change